### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by yauvink**

### CODE REVIEW

File: app/javascript/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      handleEnter(evt);
+    }
+  } else if (evt.key === 'Enter' && evt.ctrlKey) {
+    handleEnter(evt);
+  }
+
+  if (evt.key === 'Escape' && handleEscape) {
+    handleEscape(evt);
+  }
+};
```
Comment (yauvink) [inline app/javascript/utils/helpers.ts]: Can be deleted (we didn't use it anywhere)

File: app/javascript/components/create-board/create-board.tsx [hunk 2]
```diff
@@ -42,18 +75,18 @@ export const CreateBoard = ({isCreateBoardOpen, setCreateBoardOpen}) => {
...
 
-  const handleSubmit = (event) => {
+  const handleSubmit = (event: any) => {  //!!!! type
     alert(JSON.stringify(formData));
     event.preventDefault();
     setCreateBoardOpen(false);
   };
 
-  const onEmojiClick = (name) => (event, emojiObject) => {
+  const onEmojiClick = (name: string) => (event: React.MouseEvent<Element, MouseEvent>, emojiObject: IEmojiData) => {
```
Comment (yauvink) [inline app/javascript/components/create-board/create-board.tsx]:
```suggestion
  const onEmojiClick = (name: string) => (event: MouseEvent, emojiObject: IEmojiData) => {
```

onEmojiClick on line 105 gives error and wants event: MouseEvent, do you have the same?

File: app/javascript/components/create-board/create-board.tsx [hunk 3]
```diff
@@ -42,18 +75,18 @@ export const CreateBoard = ({isCreateBoardOpen, setCreateBoardOpen}) => {
...
   });
 
-  const [emojiData, setEmojiData] = useState({
+  const [emojiData, setEmojiData] = useState<{ emojiColumnName: string, emojiPickerOpen: boolean }>({
     emojiColumnName: '',
     emojiPickerOpen: false
   });
 
-  const handleSubmit = (event) => {
+  const handleSubmit = (event: any) => {  //!!!! type
```
Comment (yauvink) [inline app/javascript/components/create-board/create-board.tsx]: event: React.FormEvent can be used there

File: app/javascript/components/create-board/create-board.tsx [hunk 4]
```diff
@@ -19,13 +47,18 @@ const EmojiIcon = ({emojiColumnName, formData, setEmojiData}) => {
       {formData[emojiColumnName] ? (
         formData[emojiColumnName].emoji
       ) : (
-        <FontAwesomeIcon icon={faSmile} size="lg" />
+        <FontAwesomeIcon icon={faPropIcon} size="lg" />
```
Comment (yauvink) [inline app/javascript/components/create-board/create-board.tsx]: icon={faSmile as IconProp}

File: app/javascript/components/create-board/create-board.tsx [hunk 5]
```diff
@@ -1,11 +1,39 @@
...
+      emoji: string;
+    };
+    thirdColumnName: string;
+    thirdColumnEmoji: {
+      emoji: string;
+    };
+  }
+}
+
+const faPropIcon = faSmile as IconProp;
```
Comment (yauvink) [inline app/javascript/components/create-board/create-board.tsx]: icon={faSmile as IconProp}

File: app/javascript/components/boards-container.tsx [hunk 6]
```diff
@@ -55,10 +58,10 @@ const BoardsContainer = () => {
         }}
       >
         <FontAwesomeIcon
-          icon={faPlus}
+          icon={faPropIcon}
```
Comment (yauvink) [inline app/javascript/components/boards-container.tsx]: icon={faPlus as IconProp}

File: app/javascript/components/boards-container.tsx [hunk 7]
```diff
@@ -1,20 +1,24 @@
...
-  const [startBoards, setStartBoards] = useState([]);
-  const [historyBoards, setHistoryBoards] = useState([]);
+const BoardsContainer: React.FC = () => {
+  const [startBoards, setStartBoards] = useState<Array<BoardType> | []>([]);
+  const [historyBoards, setHistoryBoards] = useState<Array<BoardType> | []>([]);
   const [isModal, setIsModal] = useState(false);
   const [role, setRole] = useState('creator');
   const [isCreateBoardOpen, setCreateBoardOpen] = useState(false);
 
+  const faPropIcon = faPlus as IconProp;
```
Comment (yauvink) [inline app/javascript/components/boards-container.tsx]: can be deleted and changed 
icon={faPlus as IconProp}

File: app/javascript/components/board/board.tsx [hunk 8]
```diff
@@ -1,12 +1,22 @@
...
+import { BoardType } from "../../typings/board";
 import { getDate } from "../../utils/get-date";
 import { boardApi } from "../api/boards-api";
 import { Avatar } from "../avatar/avatar";
 import { MenuIcon } from "./menu-icon/menu-icon";
 import style from "./style.module.less";
 
-const Board = ({
+type Props = {
+  setHistoryBoards: (boards: Array<BoardType>) => void
```
Comment (yauvink) [inline app/javascript/components/board/board.tsx:11]: boards: BoardType[]
можно записывать еще и таким способом(но это как больше нравится и легче читается)

File: app/javascript/components/settings-page/settings-page.tsx [hunk 9]
```diff
@@ -73,13 +85,13 @@ export const SettingsPage = () => {
           <button
             type="button"
             className={style.avatarUploadButton}
-            onClick={() => fileInput.current.click()}
+            onClick={() => fileInput.current?.click()}
           >
             <FontAwesomeIcon
-              icon={faPlus}
+              icon={faPlusIcon}
```
Comment (yauvink) [inline app/javascript/components/settings-page/settings-page.tsx]: faPlus as IconProp

File: app/javascript/components/user-menu/user-menu.tsx [hunk 10]
```diff
@@ -38,7 +42,7 @@ export const UserMenu = () => {
             }}
           >
             <FontAwesomeIcon
-              icon={faCog}
+              icon={faCogIcon}
```
Comment (yauvink) [inline app/javascript/components/user-menu/user-menu.tsx]: as IconProp

File: app/javascript/components/user-menu/user-menu.tsx [hunk 11]
```diff
@@ -53,7 +57,7 @@ export const UserMenu = () => {
             }}
           >
             <FontAwesomeIcon
-              icon={faSignInAlt}
+              icon={faSignInAltIcon}
```
Comment (yauvink) [inline app/javascript/components/user-menu/user-menu.tsx]: as IconProp

File: app/javascript/utils/helpers.ts [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+  );
+};
+
+export const getBigFirstLettersBoard = (string: string) => {
+  return string.toUpperCase().charAt(0) + string.toUpperCase().charAt(1);
+};
+
+export const getInitialsTitleBoard = (title: string) => {
+  return title ? getBigFirstLettersBoard(title) : '';
+};
```
Comment (yauvink) [inline app/javascript/utils/helpers.ts]: Can be deleted

File: app/javascript/components/create-board/create-board.tsx (hunk 2 above)
Comment (gaev-art) [reply app/javascript/components/create-board/create-board.tsx]: No, I do not have this error

File: app/javascript/components/create-board/create-board.tsx [hunk 13]
```diff
@@ -85,19 +67,20 @@ export const CreateBoard: React.FC<CreateBoardProps> = ({ isCreateBoardOpen, set
...
+        emojiPickerOpen: false,
+      });
+    };
 
   return (
-    <div className={isCreateBoardOpen ? 'modal-visible' : 'modal-hidden'}>
+    <div className={isCreateBoardOpen ? "modal-visible" : "modal-hidden"}>
       <div className={style.formWrapper}>
         {emojiData.emojiPickerOpen && (
           <div className={style.emojiPickerWrapper}>
```
Comment (gaev-art) [inline app/javascript/components/create-board/create-board.tsx:86]: Type '(event: MouseEvent, emojiObject: IEmojiData) => void' is not assignable to type
 '(event: MouseEvent<Element, MouseEvent>, data: IEmojiData) => void'.

File: app/javascript/typings/emoji.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export type EmojiDataStateType = {
+  emojiColumnName: string;
+  emojiPickerOpen: boolean;
+};
+
+export type EmojiIconProps = {
+  emojiColumnName: EmojiColumnNameType;
+  setEmojiData: ({}: EmojiDataStateType) => void;
```
Comment (gaev-art) [inline app/javascript/typings/emoji.ts:17]: image

File: app/javascript/components/create-board/create-board.tsx [hunk 15]
```diff
@@ -85,19 +67,20 @@ export const CreateBoard: React.FC<CreateBoardProps> = ({ isCreateBoardOpen, set
...
-      ...formData,
-      [name]: emojiObject
-    });
-    setEmojiData({
-      emojiColumnName: '',
-      emojiPickerOpen: false
-    });
-  };
+  const onEmojiClick =
+    (name: string) => (event: MouseEvent, emojiObject: IEmojiData) => {
```
Comment (gaev-art) [inline app/javascript/components/create-board/create-board.tsx:71]: image

**Review by gaev-art — APPROVED** [review]

**Pull request merged**